packages/gitbook/src/components/TableOfContents/TOCScroller.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+import { useHash } from '../hooks';
+
+const TOCScrollContainerContext = React.createContext<React.RefObject<HTMLDivElement> | null>(null);
+
+export function TOCScrollContainerProvider(props: {
```

Comment:
```suggestion
export function TOCScrollContainer(props: {
```
It provides context but also display an element, so it's a container, not a provider.